store.go

```diff
@@ -89,7 +93,11 @@ type cidProvider interface {
 
 // Put marshals and writes content `v` to the backing blockstore returning its CID.
 func (s *BasicIpldStore) Put(ctx context.Context, v interface{}) (cid.Cid, error) {
-	mhType := uint64(mh.BLAKE2B_MIN + 31)
+	mhType := DefaultMultihash
+	if s.DefaultMultihash != 0 {
```

Comment:
Hm.... this technically conflicts with the identity hash. I mean, that doesn't make sense as the default anyways, so maybe we're fine.

Reply:
I guess I could make it a pointer?

Reply:
though if you *really* want the identity hash you can use the cid provider thing right?

Reply:
Yep. And, honestly, you shouldn't make the default "identity". That's just absolutely stupid.